modules/gdscript/gdscript_parser.cpp

```diff
@@ -3570,6 +3571,20 @@ bool GDScriptParser::export_annotations(const AnnotationNode *p_annotation, Node
 			variable->export_info.hint_string = hint_prefix + ":" + variable->export_info.hint_string;
 			variable->export_info.type = Variant::ARRAY;
 		}
+
+		if (p_annotation->name == SNAME("@export_action")) {
+			variable->export_info.hint_string = "action";
+			if (!p_annotation->resolved_arguments.is_empty()) { // Icon Name
+				variable->export_info.hint_string += (String)p_annotation->resolved_arguments[0];
+			}
+			variable->export_info.usage = PROPERTY_USAGE_EDITOR;
+		} else {
+			if (variable->export_info.type == Variant::Type::CALLABLE) {
+				push_error(R"(Export type cannot be of type Callable.)", p_annotation);
```

Comment:
We could check if the script is a tool script, and add 'Did you mean @export_action?' if true.